packages/analytics-js/src/services/StoreManager/StoreManager.ts

```diff
@@ -70,12 +70,22 @@ class StoreManager implements IStoreManager {
    * Create store to persist data used by the SDK like session, used details etc
    */
   initClientDataStore() {
-    let storageType: StorageType | '' = '';
-    storageType = state.storage.type.value || '';
+    let storageType = state.storage.type.value || '';
 
     switch (storageType) {
       case COOKIE_STORAGE:
-        if (!getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
+        // First try setting the storage to cookie else to local storage
+        if (getStorageEngine(COOKIE_STORAGE)?.isEnabled) {
+          storageType = COOKIE_STORAGE;
+        } else if (getStorageEngine(LOCAL_STORAGE)?.isEnabled) {
+          this.logger?.warn(
+            STORAGE_UNAVAILABLE_WARNING(STORE_MANAGER, COOKIE_STORAGE, LOCAL_STORAGE),
```

Comment:
We should do this for other scenarios, also. You can store the storage type in a secondary variable and compare it with the input to warn the user at the end of this switch case.

Reply:
Addressed